src/pretix/base/payment.py

```diff
@@ -190,16 +200,22 @@ def payment_form(self, request: HttpRequest) -> Form:
         form.fields = self.payment_form_fields
         return form
 
+    def _is_still_available(self, now_dt=None):
+        now_dt = now_dt or now()
+        tz = pytz.timezone(self.event.settings.timezone)
+        availability_date = self.settings.get('_availability_date', as_type=date)
+        if availability_date:
+            return availability_date >= now_dt.astimezone(tz).date()
+        return True
+
     def is_allowed(self, request: HttpRequest) -> bool:
         """
         You can use this method to disable this payment provider for certain groups
         of users, products or other criteria. If this method returns ``False``, the
         user will not be able to select this payment method. This will only be called
         during checkout, not on retrying.
-
-        The default implementation always returns ``True``.
```

Comment:
One more thing: How about instead of removing this line (and the similar one above) replacing it with something like "The default implementation checks for the ``_availability_date`` setting to be either unset or in the future.". This helps developers only reading the documentation to know how they should override this method, if they do.


We should also mention this field in the docstring of ``settings_form_fields`` for the same purpose.